sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -178,7 +179,7 @@ namespace Azure { namespace Core {
           if (typeid(T) != ptr->ValueType)
           {
             // type mismatch
-            std::abort();
+            return false;
```

Comment:
`false` has a different meaning. It is for when the key is not found.  In this case, the key is found but there is a logical/semantic error on the application that needs to be fixed (and can be fixed).  We must abort here to let people fix the bug in the code.